### PR TITLE
First set of contact tests converted to tensor mechanics

### PR DESCRIPTION
**Pull request by acasagran**
First set of tests updated for conversion to tensor mechanics. Patch tests
converted directly except for single point test. Updated SM version to use
nonlinear formulation and re-golded results. Then TM version using finite
strain matches SM results. Same approach used for Hertz heavy tests. Refs #8295.

### CODE REVIEW

**Comment by bwspenc** [comment]
@acasagran This confirms our suspicion that there is a general problem with the combination of tensor_mechanics and contact on PETSc 3.7.4 because we're seeing that same error.  I guess we can use the broken test to debug that issue, and we need to take care of that before we merge this.

**Comment by acasagran** [comment]
@bwspenc I compiled a version using PETSc 3.7.4 (clang) locally on my machine and the contact regression tests all pass. Not sure how to replicate the error. I can try the gcc version.

**Comment by bwspenc** [comment]
@acasagran I just built against PETSc 3.7.4 on my mac and the tests all pass there too.  These particular tests must only fail on Linux.

**Comment by bwspenc** [comment]
Unfortunately, these tests also fail on PETSc 3.7.5.  This seems to be an issue with the version of SuperLU that ships with that version of PETSc. If that's the case, that would put us in the tough spot where we've been before where we have to advise BISON users to use an older version of PETSc than the one that is used by MOOSE by default.

I wonder if it's something caused by the conditioning of these models, which might be remedied by changing the penalty factor.

Maybe we should just convert over the subset of the tests that works currently.

**Comment by acasagran** [comment]
I just compiled a 64-bit version using civet and all the tests passed. So, I'm going ahead and pushing again to see if it will go through the testing system. I couldn't build an intel version, so I don't know if the diffs there will happen again or not.

**Comment by permcody** [comment]
We merged in a critical TestHarness bug yesterday afternoon. The TestHarness is not reporting Diffs in HEAD right now. We are in the process of reverting but your testing is invalid. Sorry for the trouble.

On another note, we are implementing TestHarness testing immediately...

**Comment by acasagran** [comment]
I skipped a couple of tests in order to get the Intel build to pass. I turned off the volumetric locking correction in another test to get it to pass in the 64-bit build. I'm hoping those are the only changes needed to merge this PR.

File: modules/combined/tests/contact_verification/patch_tests/ring_3/ring3_mu_0_2_kin.i [hunk 1]
```diff
@@ -1,3 +1,9 @@
+[GlobalParams]
+  order = SECOND
+  family = LAGRANGE
+  volumetric_locking_correction = false
```
Comment (bwspenc) [inline modules/combined/tests/contact_verification/patch_tests/ring_3/ring3_mu_0_2_kin.i]: I assume that this is the case you referred to where you turned volumetric locking off to get it to pass with 64 bit indices.  That's not how I'd like to address that issue.  I assume this fails in a way similar to the other tests.  Let's set volumetric_locking_correction = true here, and then skip the test, with a comment that it fails on that platform.

**Review by bwspenc — CHANGES_REQUESTED** [review]
Just a couple of little things to change here.  

Do we have an issue for the problems we're having with the buggy SuperLU in PETSc 3.7.4?  If not, we need to create one, and then the few 'skip' lines we have in the tests because of that need to reference that issue.

**Review by bwspenc — APPROVED** [review]

**Comment by brianmoose** [comment]
Looks like we should have added the heavy tests to this PR and #8407. https://www.moosebuild.org/job/65144/

**Comment by bwspenc** [comment]
@brianmoose Yes, we should have -- it didn't occur to me. @acasagran -- these all look like pretty minor diffs.  Can you take care of these tests?

**Pull request merged**